src/rosdiscover/observer/ros1_actions.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+__all__ = ("separate_topics_from_action", "ActionCandidate")
+
+import re
+from typing import Collection, Dict, Optional
+
+import attr
+from loguru import logger
+from roswire import ROS1
+
+from ..core import Topic
+from ..interpreter import NodeContext
+
+_GOAL = re.compile(r'(.*)/goal')
+_GOAL_FMT = re.compile(r'(.*)Goal')
+_CANCEL = re.compile(r'(.*)/cancel')
+_STATUS = re.compile(r'(.*)/status')
+_FEEDBACK = re.compile(r'(.*)/feedback')
+_FEEDBACK_FMT = re.compile(r'(.*)Feedback')
+_RESULT = re.compile(r'(.*)/result')
+_RESULT_FMT = re.compile(r'(.*)Result')
+
+
+@attr.s(slots=True, auto_attribs=True)
+class ActionCandidate:
```

Comment:
I'm not 100% sure what it means for something to be an `ActionCandidate`. Can something be a candidate but not actually be a client or server?